src/github/client/github-client-errors.ts

```diff
@@ -121,7 +121,7 @@ export class GithubClientGraphQLError extends GithubClientError {
 		this.isRetryable = !!errors.find(
 			(error) =>
 				"MAX_NODE_LIMIT_EXCEEDED" == error.type ||
-				error.message?.startsWith("Something went wrong while executing your query")
+				error.message.startsWith("Something went wrong while executing your query")
```

Comment:
The consumer of this constructor is like
```
		const graphqlErrors = response.data?.errors;
		if (graphqlErrors?.length) {
			const err = new GithubClientGraphQLError(response, graphqlErrors);
```

I'd say a bit of defensive programming using `error.message?` is better. Cause the `errors` comes from GitHub graphql response, which isn't really validated to any format.

Reply:
I might misunderstand but this should map to GraphQLError object type where this member should always be set

Reply:
`this member should always be set` <-- I think this is the point where we weren't on the same page yet.

I am not sure about the percentage, maybe 90% of the cases above `this member should always be set` is what is happening in the real world. 

So for the rest 10% case ( or even less ), when that GraphQL errors doesn't contains the `message` property, and error happen:
1. The current behaviour is the process is treated as not `isRetryable`, and the error handling flow continue.
2. Whereas in the new behaviour, the process will throw an Error in this error handling flow, and leads to some unexpected/coded behaviour. <-- this is what I concern, and not sure what's the impact to users.

------

While majority of the errors above PR is okay, I wonder do we need to consider some edge cases? Tbh I am not fuzzy about it, we can say **"Hey the errors itself are just edges cases, errors without `message` in it are just edge cases of edge cases, so we don't need to care about it"**. <-- as long as we are aware of it and agree on that, I am fine with the PR.